subsys/bluetooth/controller/ll_sw/nordic/lll/lll_scan_aux.c

```diff
@@ -584,27 +587,142 @@ static int isr_rx_pdu(struct lll_scan_aux *lll, uint8_t devmatch_ok,
+
+		/* Check if 4 nodes free, 2 will be utilized for aux PDU and
+		 * scan response PDU; 2 more to ensure connections have them.
+		 */
+		rx = ull_pdu_rx_alloc_peek(4);
+		if (!rx) {
+			return -ENOBUFS;
+		}
+
+		/* setup tIFS switching */
+		radio_tmr_tifs_set(EVENT_IFS_US);
+		radio_switch_complete_and_rx(lll->phy);
+
+		/* prepare the scan request packet */
+		pdu_tx = (void *)radio_pkt_scratch_get();
+		pdu_tx->type = PDU_ADV_TYPE_SCAN_REQ;
+		pdu_tx->rx_addr = pdu->tx_addr;
+		pdu_tx->len = sizeof(struct pdu_adv_scan_req);
+#if defined(CONFIG_BT_CTLR_PRIVACY)
+		lrpa = ull_filter_lll_lrpa_get(rl_idx);
+		if (lll_scan->rpa_gen && lrpa) {
+			pdu_tx->tx_addr = 1;
+			memcpy(pdu_tx->scan_req.scan_addr, lrpa->val,
+			       BDADDR_SIZE);
+		} else {
+#else
+		if (1) {
+#endif /* CONFIG_BT_CTLR_PRIVACY */
+			pdu_tx->tx_addr = lll_scan->init_addr_type;
+			memcpy(pdu_tx->scan_req.scan_addr,
+			       lll_scan->init_addr, BDADDR_SIZE);
+		}
+		memcpy(pdu_tx->scan_req.adv_addr,
+		       &pdu->adv_ext_ind.ext_hdr.data[ADVA_OFFSET],
+		       BDADDR_SIZE);
+
+		radio_pkt_tx_set(pdu_tx);
+
+		/* assert if radio packet ptr is not set and radio started tx */
+		LL_ASSERT(!radio_is_ready());
+
+		if (IS_ENABLED(CONFIG_BT_CTLR_PROFILE_ISR)) {
+			lll_prof_cputime_capture();
+		}
+
+		/* capture end of Tx-ed PDU, used to calculate HCTO. */
+		radio_tmr_end_capture();
+
+#if defined(CONFIG_BT_CTLR_GPIO_PA_PIN)
+		if (IS_ENABLED(CONFIG_BT_CTLR_PROFILE_ISR)) {
+			/* PA/LNA enable is overwriting packet end
+			 * used in ISR profiling, hence back it up
+			 * for later use.
+			 */
+			lll_prof_radio_end_backup();
+		}
+
+		radio_gpio_pa_setup();
+		radio_gpio_pa_lna_enable(radio_tmr_tifs_base_get() +
+					 EVENT_IFS_US -
+					 radio_rx_chain_delay_get(lll->phy, 1) -
+					 CONFIG_BT_CTLR_GPIO_PA_OFFSET);
+#endif /* CONFIG_BT_CTLR_GPIO_PA_PIN */
+
+		radio_isr_set(isr_tx_scan_req, lll);
+
 		ull_pdu_rx_alloc();
 
+		node_rx->hdr.type = NODE_RX_TYPE_EXT_AUX_REPORT;
+
+		ftr = &(node_rx->hdr.rx_ftr);
+		ftr->param = lll;
+		ftr->ticks_anchor = radio_tmr_start_get();
+		ftr->radio_end_us = radio_tmr_end_get() -
+				    radio_rx_chain_delay_get(lll->phy, 1);
+		ftr->rssi = (rssi_ready) ? radio_rssi_get() :
+			    BT_HCI_LE_RSSI_NOT_AVAILABLE;
+		ftr->scan_req = 1U;
+		ftr->scan_rsp = 0U;
+
+#if defined(CONFIG_BT_CTLR_PRIVACY)
+		ftr->rl_idx = irkmatch_ok ? rl_idx : FILTER_IDX_NONE;
+#endif /* CONFIG_BT_CTLR_PRIVACY */
+
+		ull_rx_put(node_rx->hdr.link, node_rx);
+		ull_rx_sched();
+
+		/* switch aux scanner state to active */
+		lll->state = 1U;
+
+		return 0;
+
+	/* Passive scanner or scan responses */
+#if defined(CONFIG_BT_CENTRAL)
+	} else if (!lll_scan->conn &&
+		   lll_scan_ext_tgta_check(lll_scan, false, false, pdu,
+					   rl_idx)) {
+#else /* !CONFIG_BT_CENTRAL */
+	} else if (lll_scan_ext_tgta_check(lll_scan, false, false, pdu,
+					   rl_idx)) {
+#endif /* !CONFIG_BT_CENTRAL */
+
 		trx_cnt++;
 
+		ull_pdu_rx_alloc();
```

Comment:
This can't fail since you have done the peak above, right? 

```suggestion
		(void)ull_pdu_rx_alloc();
```